cluster/manifests/kube-state-metrics/deployment.yaml

```diff
@@ -24,17 +24,32 @@ spec:
       priorityClassName: system-cluster-critical
       serviceAccountName: kube-state-metrics
       containers:
-      - image: registry.opensource.zalan.do/teapot/kube-state-metrics:v1.3.0
-        name: kube-state-metrics
+      - name: kube-state-metrics
+        image: registry.opensource.zalan.do/teapot/kube-state-metrics:v1.8.0
         ports:
         - containerPort: 8080
-        command:
-        - /kube-state-metrics
-        - --port=8080
+          name: http-metrics
+        - containerPort: 8081
+          name: telemetry
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 8081
+          initialDelaySeconds: 5
+          timeoutSeconds: 5
+        livenessProbe:
```

Comment:
Why livenessprobe?
It will lead to metric drops

Reply:
That's from [the docs](https://github.com/kubernetes/kube-state-metrics/blob/master/examples/standard/deployment.yaml#L22-L27). I guess the idea is to kill it as soon as it's unhealthy.

Should I rather remove it? It might just stay running in unready state forever.

Reply:
On second look: It's going to a different endpoint. The [healthz](https://github.com/kubernetes/kube-state-metrics/blob/master/main.go#L246-L249) endpoint indicates the process is running and `/metrics` can be served.

Is there a reason not to kill it in those cases?

Reply:
Dropped the `liveness` probe.